src/background.js

```diff
@@ -174,7 +174,10 @@ async function onBehaviorRequested(query) {
         currentTip = TIPS.UPDATE_WEB;
         break;
     }
-  } else if (topDocIDs.has("clear")) {
+  } else if (
+    topDocIDs.has("clear") &&
+    !(await browser.tabs.query({ active: true })).incognito
```

Comment:
doesn't tabs.query resolve to an array? IS the test passing because this throws?
And you should probably specify currentWindow: true, in case multiple windows are open.
Or, in a simpler fashion, just do !(await browser.windows.getLastFocused()).incognito

Reply:
I agree that checking incognito on the top window is better. The rest of the patch is good. r+ because Marco is already blocking on this issue.